caas/kubernetes/provider/k8s.go

```diff
@@ -1090,6 +1090,12 @@ func (k *kubernetesClient) EnsureService(
 		}
 	}
 
+	if params.Deployment.DeploymentType != caas.DeploymentStateful {
+		if workloadSpec.Service != nil && workloadSpec.Service.ScalePolicy != "" {
+			return errors.NewNotValid(nil, fmt.Sprintf("ScalePolicy is only supported for %s application", caas.DeploymentStateful))
```

Comment:
This check should also be done:
- at deploy time to reject an invalid deployment immediately
- when parsing and validating charm metadata (in charm.v6)

Reply:
At that time, podspec isn't set yet.

Reply:
ah right, sorry,  i had thought that was in the charm metadata